### PR TITLE
Kneth/allow   partial in urls

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes https://github.com/realm/realm-studio/issues/658

## ☑️ ToDos

* ~[ ] 📝 Changelog entry~
* [x] 🚦 Tests
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: src/js_sync.hpp [hunk 1]
```diff
@@ -823,7 +823,13 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
             is_partial = Value::validated_to_boolean(ctx, partial_value);
         }
 
-        config.sync_config = std::make_shared<SyncConfig>(shared_user, std::move(raw_realm_url));
+        bool force_open_realms = false;
+        ValueType force_open_realms_value = Object::get_property(ctx, sync_config_object, "forceOpenRealms");
```
Comment (bdash) [inline src/js_sync.hpp]: This seems like an awfully general configuration option name for such a specific purpose.

Comment (kneth) [reply src/js_sync.hpp]: True, maybe `allowOpenPartialSyncURLs` is better.

Comment (kneth) [reply src/js_sync.hpp]: I have on purpose not updated `CHANGELOG.md` or API doc. @kraenhansen isn't happy about underscores.

Comment (bmunkholm) [reply src/js_sync.hpp]: We can update CHANGELOG.md under #internal. I think it's fine to have some way of making unsupported internal APIs clear - e.g. with _. Do we have others like that?

**Comment by bmunkholm** [comment]
@kneth I liked your other naming suggestion better. "checks" is unclear. "_allowOpeningPartialSyncUrl" is more descriptive. Not that it's critical for internal use though.

**Comment by kneth** [comment]
@bmunkholm I agree with @bdash on this one: `_allowOpeningPartialSyncUrl` almost implies that you have to set it to true if you are using partial sync. Maybe "validation" is better than "checks".

**Comment by bmunkholm** [comment]
Never mind. But perhaps a comment next to the field "Private - Only for internal use" might be good.

**Pull request merged**